### PR TITLE
Add back [ChromeWebStore] rating

### LINKED ISSUE #5475: Badges for Google Chrome Web store rating not working
Are you experiencing an issue with...

- [x] [shields.io](https://shields.io/#/)
- [ ] My own instance
- [ ] [badge-maker NPM package](https://www.npmjs.com/package/badge-maker)

:beetle: **Description**

Google Chrome Web Store rating badges seems not working since a few weeks ago.

:link: **Link to the badge**

https://img.shields.io/chrome-web-store/rating-count/oenllhgkiiljibhfagbfogdbchhdchml.svg

https://img.shields.io/chrome-web-store/rating/oenllhgkiiljibhfagbfogdbchhdchml.svg

https://img.shields.io/chrome-web-store/stars/oenllhgkiiljibhfagbfogdbchhdchml.svg

### PR DESCRIPTION
**Pull request by crimx**
Replace chrome-web-store-item-property with [webextension-store-meta](https://www.npmjs.com/package/webextension-store-meta).

fix #5475

### CODE REVIEW

**Comment by PyvesB** [comment]
Thanks for the contribution @crimx! Do you have a feel of how webextension-store-meta compares to chrome-web-store-item-property in terms of performance? HTML scrapping can be quite expensive, we need to more cautious when changing this specific family of badges.

**Comment by crimx** [comment]
You are right. Currently it's about 30 times slower. Let me see what I can do.

**Comment by crimx** [comment]
webextension-store-meta has been refactored with htmlparser2 and cheerio. Performance is now the same as chrome-web-store-item-property. Average 7.8ms per page.

**Comment by PyvesB** [comment]
> 
> 
> webextension-store-meta has been refactored with htmlparser2 and cheerio. Performance is now the same as chrome-web-store-item-property. Average 7.8ms per page.

Thanks for looking into those numbers and updating the pull request!

One thing I noticed is that `require('webextension-store-meta/lib/chrome-web-store')` takes significantly longer to execute than `require('chrome-web-store-item-property')` when the server starts up. The cheerio dependency seems to be quite heavy. Here's the kind of flame graph I get when I start Shields locally:
Screenshot 2020-10-14 221049

The big red rectangle in the middle-bottom part of the graph corresponds to the Chrome badges, roughly 15% of total start up time of the entire Shields.io app, between 0.5 and 1 seconds. My Windows machine is quite old and is not necessarily representative of what the servers actually run on, so it may be useful for someone else to pull out some numbers.

If it's only this single family of badges and the hit is only at startup, it's not the end of the world in my personal opinion, but I thought I would point this out in case there are easy wins for you to look into. Other maintainers may also have stronger feelings about this, so I'll let them weigh in as well.

**Comment by crimx** [comment]
Thanks for the info. The cheerio does seem quite heavy. Let me see if I can get rid of it.

**Comment by calebcartwright** [comment]
> Other maintainers may have stronger feelings about this though, so I'll let them weigh in as well.

I think this goes without saying but worth repeating anyway, that we object to html scraping as an approach to get data for badges. As such in the abstract this feels like a step in the wrong direction. However, given that these were only recently deprecated and the other badges in the family use the same technique, I won't object provided the change isn't negatively impactful on our runtime.

Our runtime utilizes small, resource constrained nodes that can be scaled horizontally, so in addition to the execution time performance discussed above, I would also like to be able to have a reasonable degree of confidence that some feasible X number of these badge requests hitting a node wouldn't negatively impact the environment.

**Comment by crimx** [comment]
Cheerio and other redundant dependencies has been removed. I've also added lazy parsing and cache. Parsing speed is now down to 2ms per page.

image

Comparing to chrome-web-store-item-property:

image

**Comment by PyvesB** [comment]
Looks much better! Busy end of the week for me, so I'll aim at doing a more proper review during the weekend if another maintainer doesn't pick it up in the meantime. :)

File: services/chrome-web-store/chrome-web-store-base.js
Comment (PyvesB) [inline services/chrome-web-store/chrome-web-store-base.js:3]: For consistency with the other Chrome classes, it's probably best to keep it as lower camelcase, i.e. `chromeWebStore `.

File: services/chrome-web-store/chrome-web-store-rating.service.js
Comment (PyvesB) [inline services/chrome-web-store/chrome-web-store-rating.service.js:18]: Good to see this come back! 🎉

File: services/chrome-web-store/chrome-web-store-rating.service.js
Comment (PyvesB) [inline services/chrome-web-store/chrome-web-store-rating.service.js:9]: Could you please convert this to static fields? See #5555 for more information.

**Review by PyvesB — COMMENTED** [review]
I've double-checked performance at startup and when generating badges dynamically, nothing stands out anymore. Things actually seem faster than before, so thanks for your efforts on the optimisation side. 👍🏻 

The rest of the implementation looks good to me. I've left a couple of minor comments for you to address, but we're basically there. 😉

File: services/chrome-web-store/chrome-web-store-base.js
Comment (crimx) [reply services/chrome-web-store/chrome-web-store-base.js:3]: > For consistency with the other Chrome classes

What does it mean? The uppercase is because `ChromeWebStore` is a class. `ChromeWebStore.load` is a static method.

File: services/chrome-web-store/chrome-web-store-rating.service.js
Comment (crimx) [reply services/chrome-web-store/chrome-web-store-rating.service.js]: @PyvesB Done.

File: services/chrome-web-store/chrome-web-store-base.js
Comment (PyvesB) [reply services/chrome-web-store/chrome-web-store-base.js:3]: I think I got a little confused. Let's ignore this one!

**Review by PyvesB — APPROVED** [review]

**Pull request merged**
